File: src/main/java/pl/yalgrin/playnite/simplesync/domain/Change.java
package pl.yalgrin.playnite.simplesync.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;
import pl.yalgrin.playnite.simplesync.enums.ObjectType;

import java.time.Instant;

@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
@Table("playnite_change")
public class Change {
    @Id
    @Column("id")
    private Long id;

    @Column("type")
    private ObjectType type;

    @Column("client_id")
    private String clientId;

    @Column("object_id")
    private Long objectId;

    @Column("created_at")
    private Instant createdAt;

    @Transient
    private boolean notifyAll;
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/change/domain/Change.kt
package pl.yalgrin.playnite.simplesync.change.domain

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Transient
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import pl.yalgrin.playnite.simplesync.common.enums.ObjectType
import java.time.Instant

@Table("playnite_change")
data class Change(
    @Id
    @Column("id")
    var id: Long? = null,

    @Column("type")
    var type: ObjectType,

    @Column("client_id")
    var clientId: String,

    @Column("object_id")
    var objectId: Long,

    @Column("created_at")
    var createdAt: Instant = Instant.now(),

    @Transient
    var notifyAll: Boolean = false
)
